### ISSUE TITLE
PyO3 guide docs for asyncio

### ISSUE THREAD

**Issue by davidhewitt** [PyO3/pyo3-async-runtimes#14]
PyO3 has some very out-of-date documentation for `pyo3-asyncio` at https://pyo3.rs/v0.22.5/ecosystem/async-await

We should update that documentation to reference this fork, and decide more generally what we want to do with that documentation.

**Comment by Cheukting** [comment]
@davidhewitt I am happy to work on this as I am fighting to get a workshop about async with PyO3. I have discovered many small details that would be good to have in the documentation. 

One question, for the current page https://pyo3.rs/v0.22.5/ecosystem/async-await is there any information that you want to keep or shall we start brand new?

**Comment by kylebarron** [comment]
Much of that existing docs page is still relevant, no?

**Comment by Cheukting** [comment]
Yes i wanna double check which part is still relevant, because using this fork a lot seems not needed anymore

**Comment by kylebarron** [comment]
I think except for the sections referencing pyo3-asyncio's old behavior in version 0.13, pretty much everything is still valid.

**Comment by Cheukting** [comment]
Ah sorry, I was thinking about https://pyo3.rs/v0.23.4/async-await.html and that was not very helpful to me. https://pyo3.rs/v0.23.4/ecosystem/async-await.html is similar to README here and yes it is much more useful.

So let me try asking the question again. Do we still want the https://pyo3.rs/v0.23.4/async-await.html  page or just direct folks to use this fork with the guide at https://pyo3.rs/v0.23.4/ecosystem/async-await.html ?

**Comment by kylebarron** [comment]
I believe the confusion is that these two docs pages document **two separate approaches** for managing async with Rust. https://pyo3.rs/v0.23.4/async-await.html documents the _native_ support in pyo3, while the ecosystem page documents the use of an external crate (i.e. this `pyo3-async-runtimes` crate) to handle the mapping.

This crate is the older approach, with slightly more overhead. While I think the native support is thought to be better in the long term but maybe not fully ready yet. For reference in my projects (e.g. https://github.com/developmentseed/obstore) I use this `pyo3-async-runtimes` crate.

**Comment by Cheukting** [comment]
I see, I am also using `pyo3-async-runtimes` crate. And I didn't realize there is this page: https://pyo3.rs/v0.23.4/ecosystem/async-await.html (would have been helpful if I found it earlier. I will have a look at this page and see if there's any need to update then (I think that page is the same as the README here so maybe nothing needs to be done)

**Comment by davidhewitt** [comment]
The PyO3 documentation has now been updated on `main` to point at this crate. I think there's no need to rush to do more on this issue, any documentation can live on this package.
